src/connectWaiter.js

```diff
@@ -68,6 +68,10 @@ export default (params) => {
         }
       }
 
+      componentDidCatch(error) {
```

Comment:
@seesleestak let me know what you think of all this. Been looking forward to using it to stop false "reject" states when the wrapped component throws an error.

Reply:
Besides logging the warning, do we have to do anything to handle the `error`?

Also, I'm seeing this from eslint -- which I imagine we can disable if it's not needed:
```
class-methods-use-this: Expected 'this' to be used by class method 'componentDidCatch'.                                                                     
```

Reply:
I'm also seeing it complain about the unused `error`

Reply:
This is meant to safeguard the waiter request lifecyle, so I don't think connectWaiter should handle the error. Leave that to user implementing the component. I'll see about fixing the lint errors